packages/terra-functional-testing/tests/jest/commands/expect/toMatchReference.test.js

```diff
@@ -1,6 +1,14 @@
 const toMatchReference = require('../../../../src/commands/expect/toMatchReference');
 
 describe('toMatchReference', () => {
+  beforeEach(() => {
```

Comment:
In my opinion, we should make each unit test self contained by removing this setup from the beforeEach, and inserting it into every test  - there will be duplication, but each test will be extremely clear on the setup and assertions. Otherwise, we need to remember that `ignoreScreenshotMismatch` is false by default, which is defined all the way at the top of the file. What do you think?

Reply:
I think we're better off with this defined in the beforeEach. While setting up each test individually could be clearer, as far as I am aware the current planned design for `ignoreScreenshotMismatch` is pretty simple and only really effects one specific code decision point. If this was handing off an object or something capable of more variance I feel like setting it up in each unit test could be a better solution. We only are ever going to need it to be `true` instead of `false`  in the one example jest test that has been added and in the case where it is `true` it changes a single logic point. If this touched on more logic or was planned to touch more logic (within this test file) I could see the need to make more of a case by case basis similar to how the `receivedScreenshot` object is handled in the rest of the file. Also not repeating code is always nice.

Reply:
Sure, let's follow your outline then - if we only need to test a single use case where it's set to false, and the rest of use cases are true, how about using `beforeAll`? 

Rather than re-setting a global value to false for every test, we'd use `beforeAll` to default it to false, then set and reset it within the unit test that needs it to be true. This eliminates the redundant setup that executes for every single block. What do you think?

Reply:
That works for me. Made the changes here https://github.com/cerner/terra-toolkit/pull/655/commits/ac8bd511f4e95ae21e42259858cebb7bc8e4ca95